nima_core/federation/pg_storage.py

```
@@ -0,0 +1,370 @@
+        return self._conn
+
+    def __exit__(self, exc_type, exc, tb):
+        try:
+            if exc_type is not None:
+                self._conn.rollback()
+            else:
+                # Close any implicit open transaction (e.g. from read-only queries)
+                # so the connection is not returned to the pool as "idle in transaction".
+                self._conn.rollback()
+        except Exception:
+            pass
```

Ruff (0.15.10)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)